### PR TITLE
Add backend code to get token - Kotlin

### PR DESCRIPTION
**Pull request by cristobal-buenrostro**

### CODE REVIEW

File: js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt [hunk 1]
```diff
@@ -49,15 +47,17 @@ class MainActivity : AppCompatActivity() {
...
         val disableLanguageDetectionButton = findViewById<Button>(R.id.disableLanguageDetectionButton)
         val setLanguage = findViewById<EditText>(R.id.editTextLanguage);
-        immersiveReaderButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView() } }
-        disableGrammarButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableGrammar") } }
-        disableTranslationButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableTranslation") } }
-        disableLanguageDetectionButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableLanguageDetection", setLanguage.text.toString()) } }
+        val checkBoxToken = findViewById<CheckBox>(R.id.checkBoxToken);
+        immersiveReaderButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView(tokenFromServer = checkBoxToken.isChecked) } }
+        disableGrammarButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableGrammar", tokenFromServer = checkBoxToken.isChecked) } }
```
Comment (rwallerms) [inline js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt:52]: checkBox**Token**.isChecked for all Disable options? Typo?

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt (hunk 1 above)
Comment (cristobal-buenrostro) [reply js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt:52]: I didn't get the point here.
I'm sending the checkbox value to the button handler. Each option has its own button, that's the reason you see it for all options.

File: js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt [hunk 2]
```diff
@@ -94,15 +94,23 @@ class MainActivity : AppCompatActivity() {
         var token: String
 
         runBlocking{
-            val resp = async { getImmersiveReaderTokenAsync() }
+            val resp = async { getImmersiveReaderTokenAsync(tokenFromServer) }
             token = resp.await()
-            val jsonResp = JSONObject(token)
-            loadImmersiveReaderWebView(exampleActivity, jsonResp.getString("access_token"), subdomain, content, options)
+            if (!tokenFromServer) {
```
Comment (rwallerms) [inline js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt:99]: code to get token from server should be in an else block

File: js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt (hunk 1 above)
Comment (rwallerms) [reply js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt:52]: OK

File: js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt [hunk 3]
```diff
@@ -49,15 +47,17 @@ class MainActivity : AppCompatActivity() {
         val disableTranslationButton = findViewById<Button>(R.id.disableTranslationButton)
         val disableLanguageDetectionButton = findViewById<Button>(R.id.disableLanguageDetectionButton)
         val setLanguage = findViewById<EditText>(R.id.editTextLanguage);
-        immersiveReaderButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView() } }
-        disableGrammarButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableGrammar") } }
-        disableTranslationButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableTranslation") } }
-        disableLanguageDetectionButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView("disableLanguageDetection", setLanguage.text.toString()) } }
+        val checkBoxToken = findViewById<CheckBox>(R.id.checkBoxToken);
+        immersiveReaderButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView(tokenFromServer = checkBoxToken.isChecked) } }
```
Comment (rwallerms) [inline js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt:51]: tokenFromServer = checkBoxToken.isChecked

Should create a local variable for this that you pass to all the following calls, _getTokenFromServer_

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt (hunk 2 above)
Comment (cristobal-buenrostro) [reply js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt:100]: done

File: js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt (hunk 3 above)
Comment (cristobal-buenrostro) [reply js/samples/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt]: done

**Review by rwallerms — APPROVED** [review]

**Pull request merged**